lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,716 @@
+import { getCodeCommitUrl } from './codecommit-client';
+import * as client from './codecommit-client';
+import { getUserArn, initIamClient } from './iam-client';
+
+interface Config {
+  repository?: string;
+  defaultBranch?: string;
+  region?: string;
+  prList?: Pr[];
+  credentials?: Credentials;
+  userArn?: string;
+}
+
+export const config: Config = {};
+
+export async function initPlatform({
+  endpoint,
+  username,
+  password,
+}: PlatformParams): Promise<PlatformResult> {
+  let accessKeyId = username;
+  let secretAccessKey = password;
+  let region;
+
+  if (!accessKeyId) {
+    accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  }
+  if (!secretAccessKey) {
+    secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  }
+  if (endpoint) {
+    const regionReg = regEx(/.*codecommit\.(?<region>.+)\.amazonaws\.com/);
+    const codeCommitMatch = regionReg.exec(endpoint);
+    region = codeCommitMatch?.groups?.region;
+    if (!region) {
+      logger.warn("Can't parse region, make sure your endpoint is correct");
+    }
+  } else {
+    region = process.env.AWS_REGION;
+  }
+
+  if (!accessKeyId || !secretAccessKey || !region) {
+    throw new Error(
+      'Init: You must configure a AWS user(accessKeyId), password(secretAccessKey) and endpoint/AWS_REGION'
+    );
+  }
+
+  config.region = region;
+  const credentials: Credentials = {
+    accessKeyId,
+    secretAccessKey,
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  };
+  config.credentials = credentials;
+
+  // If any of the below fails, it will throw an exception stopping the program.
+  client.buildCodeCommitClient(region, credentials);
+  // To check if we have permission to codecommit
+  await client.listRepositories();
+
+  initIamClient(region, credentials);
+  config.userArn = await getUserArn();
+
+  const platformConfig: PlatformResult = {
+    endpoint: endpoint ?? `https://git-codecommit.${region}.amazonaws.com/`,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+export async function initRepo({
+  repository,
+  endpoint,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`initRepo("${repository}")`);
+
+  config.repository = repository;
+
+  let repo;
+  try {
+    repo = await client.getRepositoryInfo(repository);
+  } catch (err) {
+    logger.error({ err }, 'Could not find repository');
+    throw new Error(REPOSITORY_NOT_FOUND);
+  }
+
+  const url = getCodeCommitUrl(config.region!, repository, config.credentials!);
+  try {
+    await git.initRepo({
+      url,
+    });
+  } catch (err) {
+    logger.debug({ err }, 'Failed to git init');
+    throw new Error(PLATFORM_BAD_CREDENTIALS);
+  }
+
+  if (!repo?.repositoryMetadata) {
+    logger.error({ repository }, 'Could not find repository');
+    throw new Error(REPOSITORY_NOT_FOUND);
+  }
+
+  logger.debug({ repositoryDetails: repo }, 'Repository details');
+  const metadata = repo.repositoryMetadata;
+
+  if (!metadata?.defaultBranch || !metadata?.repositoryId) {
```

Comment:
```suggestion
  if (!metadata.defaultBranch || !metadata.repositoryId) {
```
already validated above